src/classic/element/ReactElementValidator.js

```diff
@@ -154,17 +154,22 @@ function warnAndMonitorForKeyUse(message, element, parentType) {
   // Usually the current owner is the offender, but if it accepts children as a
   // property, it may be the creator of the child that's responsible for
   // assigning it a key.
+  var childOwnerAddendum = '';
   if (element &&
       element._owner &&
       element._owner !== ReactCurrentOwner.current) {
     // Name of the component that originally created this child.
     var childOwnerName = getName(element._owner);
 
-    message += ` It was passed a child from ${childOwnerName}.`;
+    childOwnerName = ` It was passed a child from ${childOwnerName}.`;
   }
 
-  message += ' See http://fb.me/react-warning-keys for more information.';
-  warning(false, message);
+  warning(
+    false,
+    message + '%s%s See http://fb.me/react-warning-keys for more information.',
```

Comment:
Makes me a little queasy to see a function that takes in a message and prepends it to the beginning of a format string, because it's not entirely clear from the signature that the caller of warnAndMonitorForKeyUse is not allowed to pass user-controlled data into the message.